buildflow/runtime/managers/stream_manager.py

```diff
@@ -0,0 +1,51 @@
+import ray
+
+from buildflow.api import options
+from buildflow.runtime.managers import processors
+from buildflow.runtime.ray_io import empty_io
+
+
+@ray.remote
+class _StreamManagerActor:
+
+    def __init__(self, options: options.StreamingOptions,
+                 processor_ref: processors.ProcessorRef) -> None:
+        self.options = options
+        self.processor_ref = processor_ref
+
+    def _add_replica(self):
+        key = str(self.processor_ref.sink)
+        if isinstance(self.processor_ref.sink, empty_io.EmptySink):
+            key = 'local'
+        processor_actor = processors.ProcessActor.remote(
+            self.processor_ref.get_processor_replica())
+        sink_actor = self.processor_ref.sink.actor(
+            processor_actor.process_batch.remote,
+            self.processor_ref.source.is_streaming())
+
+        source_actor = self.processor_ref.source.actor({key: sink_actor})
+        return source_actor.run.remote()
+
+    async def run(self):
+        start_replics = self.options.min_replicas
+        if self.options.num_replicas:
+            start_replics = self.options.num_replicas
+        replicas = [self._add_replica() for _ in range(start_replics)]
+        while True:
```

Comment:
There should probably be a sleep in the while loop so we dont eat up a core. Maybe a 60s sleep?